docs/rationale.md

```diff
@@ -23,4 +23,72 @@ try (Scope scope = TracingContextUtils.currentContextWith(span)) {
 ```
 
 It would not be possible to call `recordException` if `span` was also using try-with-resources.
-Because this is a common usage for spans, we do not support try-with-resources.
\ No newline at end of file
+Because this is a common usage for spans, we do not support try-with-resources.
+
+
+## Versioning and Releases
+
+### Assumptions
+
+- This project uses semver v2, as does the rest of OpenTelemetry.
+
+### Goals
+
+- API Stability: 
+    - Once the API for a given signal (spans, logs, metrics, baggage) has been officially released, code instrumented with that API module will 
+function, *with no recompilation required*, with any API+SDK that has the same major version, and equal or greater minor or patch version. 
+    - For example, libraries that are instrumented with `opentelemetry-api-trace:1.0.1` will function, at runtime with 
+SDK library `opentelemetry-sdk-trace:1.11.33` plus `opentelemetry-api-trace:1.11.33`. 
+    - We call this requirement the "ABI" compatibility requirement for "Application Binary Interface" compatibility. 
+- SDK Stability:
+    - Public portions of the SDK (constructors, configuration, und-user interfaces) must remain backwards compatible.
+        - Precisely what this includes has yet to be delineated. 
+    - Internal interfaces are allowed to break.
```

Comment:
I guess this is referring to interfaces in the SDK that have the public modifier but don't fall into the above bucket. While I know `Precisely what this includes has yet to be delineated. ` do you have any example of it? `BatchSpanProcessor`?

Reply:
I copied this out of Ted's otep...I think it just means non-public "internals" of the SDK. It might not be a valuable thing to call out.

Reply:
Ok - yeah it might be worth calling out but it applies to both API and SDK so weird for me to see it here, which I guess is my comment on the OTEP too so just repeating the feedback it seems :)

Reply:
updated